app/js/controllers/actions/event/event-list-ctrl.js

```diff
@@ -0,0 +1,299 @@
+'use strict';
+
+angular
+.module('app.controllers')
+.controller('EventListCtrl', function($scope, $sce, $rootScope, $stateParams, $state, Restangular, apiDescriptor, dataTransformer) {
+	var resourceName = $stateParams.resourceName;
+	var resourceId = $stateParams.id;
+	$scope.resourceName = resourceName;
+	apiDescriptor.then(function(apiDescription) {
+		$scope.rdesc = apiDescription.resource(resourceName);
+	});
+
+	$scope.displayDate = function(date) {
+		if (date == undefined) { return; };
+		var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+		var year = parseInt(date.substring(0,4));
+		var month = parseInt(date.substring(5, 7));
+		return monthNames[month - 1] + " " + year;
+	}
+
+	var teamsIdToName = {};
+	var venuesIdToName = {};
+	var organizationIdToName = {};
+	var personIdToName = {};
+
+	$scope.eventDetails = {
+		venue : {},
+		team 	: {},
+		survey: {},
+		feedback: {},
+		coorganizers: {},
+		presenters: {},
+		addedBy: {},
+		attendees: {},
+		attendeesCount: {},
+		teaches: {},
+		date: {},
+		rsvps: {},
+		rsvpCount: {},
+		altRsvps: {},
+		aims: {},
+		categories: {}
+	};
+
+	//mapping teamID to teamName
+	Restangular.all('teams')
+		.getList()
+		.then(function(teams) {
+			_.each(teams, function(element) {
+				teamsIdToName[element.id] = element.attributes.name;
+			});
+		});
+
+	//mapping venueID to venueName
+	Restangular.all('venues')
+		.getList()
+		.then(function(teams) {
+			_.each(teams, function(element) {
+				venuesIdToName[element.id] = element.attributes.name;
+			});
+		});
+
+	//mapping organization to organizationName
+	Restangular.all('organizations')
+		.getList()
+		.then(function(organization) {
+			_.each(organization, function(element) {
+				organizationIdToName[element.id] = element.attributes.name;
+			});
+		});
+
+	//mapping personID to personName
+	Restangular.all('people')
+		.getList()
+		.then(function(person) {
+			_.each(person, function(element) {
+				personIdToName[element.id] = element.attributes.name;
+			});
+		})
+		.then(function(){
+			Restangular.all(resourceName)
+			.getList()
+			.then(function(data) {
+				$scope.data = data;
+				if (resourceId) {
+					var index = _.findIndex($scope.data, {id: resourceId});
+					$scope.model = $scope.data[index];
+				}
+
+				_.each($scope.data, function(element) {
+					//mapping eventID to venue names with links
+					if (element.relationships.venue.data != null) {
+						// !! change venueURL to the venue page once the venue override page is ready
+						var venueURL = "https://api.tnyu.org/v3/venues/" + element.relationships.venue.data.id;
+						$scope.eventDetails.venue[element.id] = "<a href=" + venueURL + ">" + venuesIdToName[element.relationships.venue.data.id] + "</a>";
+					};//end
```

Comment:
let's remove all these `end`s will be clear off of the brackets and indentation
also, remove the `;` after if statement closing brackets